spec/call_spec.rb

```diff
@@ -0,0 +1,73 @@
+describe 'Call' do
+  let(:source) { '31621234567' }
+  let(:destination) { '31621234568'}
+  let(:call_id) { "21025ed1-cc1d-4554-ac05-043fa6c84e00" }
+  let(:http_client) { double(MessageBird::HttpClient) }
+  let(:client) { MessageBird::Client.new('', http_client) }
+  let(:call_flow) do 
+    '{ 
+      "title": "Say message", 
+      "steps": [ 
+        { 
+          "action": "say", 
+          "options": { 
+            "payload": "This is a journey into sound. Good bye!", 
+            "voice": "male", 
+            "language": "en-US" 
+          } 
+        } 
+      ] 
+    }'
+  end
+
+  it 'create a call' do
+    expect(http_client)
+      .to receive(:request)
+      .with(:post, 'calls', {source: source, destination: destination, callFlow: call_flow})
+      .and_return('{"data":[{"id":"'+call_id+'","status":"queued","source":"'+source+'","destination":"'+destination+'","createdAt":"2019-10-11T13:02:19Z","updatedAt":"2019-10-11T13:02:19Z","endedAt":null}],"_links":{"self":"/calls/'+call_id+'"},"pagination":{"totalCount":0,"pageCount":0,"currentPage":0,"perPage":0}}')
+    call = client.call_create(source, destination, call_flow)
+    expect(call.id).to eq call_id 
+  end
+
+  it 'list all calls' do
+    expect(http_client)
+      .to receive(:request)
+      .with(:get, 'calls', {})
+      .and_return('{ "data": [ { "id": "f1aa71c0-8f2a-4fe8-b5ef-9a330454ef58", "status": "ended", "source": "'+source+'", "destination": "31612345678", "createdAt": "2017-02-16T10:52:00Z", "updatedAt": "2017-02-16T10:59:04Z", "endedAt": "2017-02-16T10:59:04Z", "_links": { "self": "/calls/f1aa71c0-8f2a-4fe8-b5ef-9a330454ef58" } }, { "id": "ac07a602-dbc1-11e6-bf26-cec0c932ce01", "status": "ended", "source": "'+source+'", "destination": "31612345678", "createdAt": "2017-01-16T07:51:56Z", "updatedAt": "2017-01-16T07:55:56Z", "endedAt": "2017-01-16T07:55:56Z", "_links": { "self": "/calls/ac07a602-dbc1-11e6-bf26-cec0c932ce01" } } ], "_links": { "self": "/calls?page=1" }, "pagination": { "totalCount": 2, "pageCount": 1, "currentPage": 1, "perPage": 10 } }
+')
+    list = client.call_list
+    expect(list.items).not_to be_nil
+    expect(list.items.first.source).to eq source
+  end
+
+  it 'view a call' do
+    expect(http_client)
+      .to receive(:request)
+      .with(:get, "calls/#{call_id}", {})
+      .and_return('{
+  "data": [
+    {
+      "id": "'+call_id+'",
+      "status": "ended",
+      "source": "'+source+'",
+      "destination": "'+destination+'",
+      "createdAt": "2017-02-16T10:52:00Z",
+      "updatedAt": "2017-02-16T10:59:04Z",
+      "endedAt": "2017-02-16T10:59:04Z"
+    }
+  ],
+  "_links": {
+    "self": "/calls/f1aa71c0-8f2a-4fe8-b5ef-9a330454ef58"
+  }
+}')
+    client.call_view(call_id)
```

Comment:
Can you please check the returned call and its attributes? Right now, we're only testing that the HTTP client receives the request we expect it to, but not whether the MessageBird::Client properly deserialises the response.

Reply:
done!